zipkin-server/src/test/java/zipkin2/server/internal/prometheus/ITZipkinMetrics.java

```diff
@@ -77,7 +77,7 @@
 
     // ensure we don't track prometheus, UI requests in prometheus
     assertThat(scrape())
-      .doesNotContain("prometheus")
+      .doesNotContain("prometheus_notifications")
```

Comment:
This one as well. Any better suggestion please?

Reply:
similarly we intentionally shouldn't see any meta stats about prometheus. Possibly this is a correct glitch and something else we need to disable now. Stats cost money (especially in SaaS) and we don't want to blanket add things especially monitoring the monitor without being conscious about it.

Reply:
The metric contains `service="zipkin2.server.internal.prometheus.ZipkinMetricsController"` so it's the problem. 😄 

Here's the story.
We fixed to add FQCN of the service name for HTTP services as a metric tag in 0.99.7. https://github.com/line/armeria/pull/2780
So if the call is made to `ZipkinMetricsController.fetchMetricsFromMicrometer()`, 
one of the metric recorded is `...method="fetchMetricsFromMicrometer",service="zipkin2.server.internal.prometheus.ZipkinMetricsController",quantile="0.0"...`

But we had a bug in 0.99.8 that the `MetricCollectingService` is not automatically added.
If Armeria server is created using spring boot integration and `ArmeriaSettings.isEnableMetric()` returns `true`, `MetricCollectingService` is added by default.
https://github.com/line/armeria/pull/2898/files#diff-58a90bb70d1f88b9c2f45ee9bddd4468R126
https://github.com/line/armeria/pull/2898/files#diff-58a90bb70d1f88b9c2f45ee9bddd4468R148

So from 0.99.9 we add `MetricCollectingService` the `Server` so all the requests and responses are recorded by default.

Now, `metricsIsOK()` is executed before `prometheusIsOK()` is executed during the test which records the metric about `service="zipkin2.server.internal.prometheus.ZipkinMetricsController"` which contains `prometheus` that make the test fails.

Reply:
So it's better not to add `MetricCollectingService` by default using `ArmeriaSettings`? 🤔